test/e2e-tests.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the end-to-end tests for the kn client.
+
+# If you already have the `KO_DOCKER_REPO` environment variable set and a
+# cluster setup and currently selected in your kubeconfig, call the script
+# with the `--run-tests` argument and it will use the cluster and run the tests.
+
+# Calling this script without arguments will create a new cluster in
+# project $PROJECT_ID, start Knative serving, run the tests and delete
+# the cluster.
+
+source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh
+
+# Helper functions.
+
+function knative_setup() {
+  start_latest_knative_serving
+}
+
+
+# Script entry point.
+
+header "Building client"
```

Comment:
My apologies, this will be run twice if you don't use an `if` here. The test won't fail, but the log will be confusing. I suggest putting the build phase in a function. Something like:

```
...
function build_kn() {
   GO111MODULE=on go build -v ./cmd/... || fail_test "Error building kn"
}

# Script entry point.

# Build kn before starting knative, so we fail fast in case of error.
(( ! RUN_TESTS )) && build_kn

initialize $@
...
```

Reply:
Thanks for the pointer - I did notice after making the last change that the test output showed it trying to build the client twice, the second time with an unexpected warning message. Now I understand why.

Reply:
It looks like checking RUN_TESTS isn't the right variable - this still built twice. I can just punt on trying to get this to fail-fast and put the building part in a `test_setup` function, I guess.

Reply:
Actually RUN_TESTS is not initialized at this point. https://github.com/knative/test-infra/issues/666 is the clean way of doing this.